common/models/model-method.js

```diff
@@ -10,4 +10,43 @@ module.exports = function(ModelMethod) {
    * @class ModelMethod
    * @inherits WorkspaceEntity
    */
+
+  ModelMethod.getJsonKey = function(data) {
+    var version = ModelMethod.app.models.Workspace.loopBackVersion;
+    if (encodeStaticFlagInName(version)) {
+      var isStatic = data.isStatic;
+      if (isStatic !== undefined) {
+        var name = data.name;
+        var matchName = name.match(/^prototype\.(.*)$/);
+        if (!isStatic && (matchName === null || !matchName)) {
+          data.name = 'prototype.' + name;
+        }
+      }
+      delete data.isStatic;
+    }
+    return data.name;
+  }
+
+  ModelMethod.getConfigFromData = function(data) {
+    var version = ModelMethod.app.models.Workspace.loopBackVersion;
+    var properties = this.definition.properties;
+    var result = {};
+    var prop;
+
+    if (encodeStaticFlagInName(version)) {
+      delete data.isStatic;
+    }
+    
+    for (prop in properties) {
+      if (properties[prop].json === false) continue;
+      result[properties[prop].json || prop] = data[prop];
+    }
+
+    return result;
+  };
 };
+
+function encodeStaticFlagInName(version) {
```

Comment:
In order to avoid repetition of `ModelMethod.app.models.Workspace.loopBackVersion`, I am proposing to change this function into a private model method.

```js
ModelMethod._shouldEncodeStaticFlagInName = function() {
  var version = ModelMethod.app.models.Workspace.loopBackVersion; 
  // ...
};
```

Thoughts?